appeals/web/src/server/lib/mappers/appeal/appeal.mapper.js

```diff
@@ -0,0 +1,174 @@
+ * @property {WebAppeal} appealDetails
+ * @property {string} currentRoute
+ * @property {SessionWithAuth} session
+ * @property {boolean} [skipAssignedUsersData]
+ * @property {{
+ *     [x: string]: any;
+ *   } | null} [caseOfficerUser]
+ * @property {{
+ *     [x: string]: any;
+ *   } | null} [inspectorUser]
+ */
+
+/**
+ * @typedef {(params: SubMapperParams) => Instructions} SubMapper
+ */
+
+// Not super buzzed about these keys, is there something nicer than appealType we can use here?
+/** @type {Record<string, Record<string, SubMapper>>} */
+const submaps = {
+	Householder: hasSubmaps,
+	'Planning appeal': fpaSubmaps
+};
+
+/** @type {Mapper} */
+export async function initialiseAndMapAppealData(
+	appealDetails,
+	currentRoute,
+	session,
+	skipAssignedUsersData = false
+) {
+	if (appealDetails === undefined) {
+		throw new Error('appealDetails is undefined');
+	}
+
+	if (!appealDetails.appealType) {
+		throw new Error('No appealType on appealDetails');
+	}
+
+	const caseOfficerUser = appealDetails.caseOfficer
+		? await usersService.getUserByRoleAndId(
+				config.referenceData.appeals.caseOfficerGroupId,
+				session,
+				appealDetails.caseOfficer
+		  )
+		: null;
+
+	const inspectorUser = appealDetails.inspector
+		? await usersService.getUserByRoleAndId(
+				config.referenceData.appeals.caseOfficerGroupId,
+				session,
+				appealDetails.inspector
+		  )
+		: null;
+
+	const submappers = submaps[appealDetails.appealType];
+
+	/** @type {{appeal: MappedInstructions}} */
+	const mappedData = {
+		appeal: {}
+	};
+
+	Object.entries(submappers).forEach(([key, submapper]) => {
+		mappedData.appeal[key] = submapper({
+			appealDetails,
+			currentRoute,
+			session,
+			skipAssignedUsersData,
+			caseOfficerUser,
+			inspectorUser
+		});
+	});
+
+	return mappedData;
+}
+
+/**
+ * @param {WebAppeal} appealDetails
+ * @returns {string}
+ */
+export function generateLinkedAppealsManageLinkHref(appealDetails) {
+	const baseUrl = `/appeals-service/appeal-details/${appealDetails.appealId}/linked-appeals`;
+
+	if (appealDetails.linkedAppeals.length > 0) {
+		return `${baseUrl}/manage`;
+	}
+
+	return `${baseUrl}/add`;
+}
+
+/**
+ * @param {WebAppeal} appealDetails
+ * @returns {string}
+ */
+export function generateAppealDecisionActionListItems(appealDetails) {
+	switch (appealDetails.appealStatus) {
+		case APPEAL_CASE_STATUS.ISSUE_DETERMINATION: {
+			return `<li class="govuk-summary-list__actions-list-item"><a class="govuk-link" href="${generateIssueDecisionUrl(
+				appealDetails.appealId
+			)}">Issue</a></li>`;
+		}
+		case APPEAL_CASE_STATUS.COMPLETE: {
+			return `<li class="govuk-summary-list__actions-list-item">${generateDecisionDocumentDownloadHtml(
+				appealDetails
+			)}</li>`;
+		}
+		default: {
+			return '';
+		}
+	}
+}
+
+/**
+ * @param {WebAppeal} appealDetails
+ * @param {string} [linkText]
+ * @returns {string}
+ */
+export function generateDecisionDocumentDownloadHtml(appealDetails, linkText = 'View') {
```

Comment:
Just a heads up, I noticed Ben's refactoring PR touches this function too (https://github.com/Planning-Inspectorate/appeals-back-office/pull/492) - and probably some more besides, as they both relate to refactoring of mappers.

Reply:
I don't think I actually changed anything about this function. I'm hoping it's just green because I moved this whole file from somewhere else

Reply:
Hopefully that should make conflict resolution a bit easier!